doc/source/dev/data_types.md

```diff
@@ -0,0 +1,607 @@
+        try:
+            if len(headers) < 2:
+                return False
+            for hdr in headers:
+                if len(hdr) > 1 and hdr[0] and not hdr[0].startswith('#'):
+                    if len(hdr) != 8:
+                        return False
+                    try:
+                        map(int, [hdr[6], hdr[7]])
+                    except:
+                        return False
+                # Do other necessary checking here...
+        except:
+            return False
+        # If we haven't yet returned False, then...
+        return True
+
+    ...
+```
+
+That should be it! If all of your code is 
+functionally correct you should now have support for 
+your new data type within your Galaxy instance.
+
+## Adding a New Data Type (completely new)
+
+### Basic Datatypes
+In this [real life example](https://github.com/bgruening/galaxytools/blob/master/datatypes/common_sequence_datatypes/csequence.py), 
+we'll add a datatype named `GenBank`, to support 
+GenBank files. 
+
+First, we'll set up a file named `csequence.py` in 
+`lib/galaxy/datatypes/csequence.py`. This file could 
+contain some of the standard sequence types, though 
+we'll only implement GenBank.
+
+```python
+"""
+Classes for all common sequence formats
+"""
+
+from galaxy.datatypes import data
+from galaxy.datatypes.metadata import MetadataElement
+
+import os
+import logging
+
+log = logging.getLogger(__name__)
+
+class GenBank( data.Text ):
+    """
+        abstract class for most of the molecule files
+    """
+    file_ext = "genbank"
+```
+
+This is all you need to get started with a datatype. 
+Now, load it into your `datatypes_conf.xml` by adding 
+the following line:
+
+```xml
+<datatype extension="genbank" type="galaxy.datatypes.csequence:GenBank" display_in_upload="True" />
+```
+
+and start up your server. Were you watching the logs 
+carefully? No? Wondering why your module isn't showing 
+up in the upload tool? Well, if you dig through your 
+logs you'll see this message:
+
+```
+galaxy.datatypes.registry ERROR 2014-07-17 12:43:23,939 Error importing datatype module galaxy.datatypes.csequence: 'module' object has no attribute 'csequence'
+Traceback (most recent call last):
+  File "/home/hxr/work/galaxy-central/lib/galaxy/datatypes/registry.py", line 208, in load_datatypes
+    module = getattr( module, mod )
+AttributeError: 'module' object has no attribute 'csequence'
+```
+
+This error comes as a result of the module not being 
+imported by `registry.py`. You'll need to add your 
+module as an import to the top of `registry.py`:
+
+```python
+import csequence
+```
+
+Once you've done this, your server will start up and 
+the datatype will be available. Please note that this 
+problem can be avoided by using the Tool Shed to store 
+your datatypes. There, this issue will be avoided as 
+Galaxy handles imports from Tool Shed installed 
+datatypes differently than from locally installed 
+datatypes.
```

Comment:
Is this still the case? I think all that's needed any more since commit ce54045b20996584f1b83a1966d4f526f3f4fdf2 (from 2014).